### PR TITLE
Invoke callback URL when build has failed

### LINKED ISSUE #522: Callback URL doesn't invoked when build fails
[Callback URL section](https://github.com/openshift/source-to-image/blob/master/docs/cli.md#callback-url) a bit outdated:
- `labels` field is missing
- description of `success` says that this field `indicating the result of the build process (true or false)` but this callback never gets executed if build failed. It only executed for a successful builds and hence this field always has value `true` (I'm not sure whether it's a bug in the code or documentation)

### PR DESCRIPTION
**Pull request by php-coder**
Fixes #522

PTAL @bparees

### CODE REVIEW

File: pkg/util/callback.go [hunk 1]
```diff
@@ -34,16 +34,19 @@ func (c *callbackInvoker) ExecuteCallback(callbackURL string, success bool, labe
...
 
-	d := map[string]interface{}{
-		"labels":  labels,
+	data := map[string]interface{}{
 		"payload": buf.String(),
 		"success": success,
 	}
 
+	if len(labels) > 0 {
+		data["labels"] = labels
```
Comment (php-coder) [inline pkg/util/callback.go]: With this check when build failed we don't send `labels` field at all (before it was `null`).

File: pkg/build/strategies/sti/postexecutorstep_test.go [hunk 2]
```diff
@@ -235,49 +235,3 @@ func TestReportSuccessStep(t *testing.T) {
...
-	}
-
-	if !reflect.DeepEqual(callbackInvoker.Messages, expectedMessages) {
-		t.Errorf("should invoke ExecuteCallback(Messages=%v) but invoked with %v", expectedMessages, callbackInvoker.Messages)
-	}
-
-	if !reflect.DeepEqual(callbackInvoker.Labels, expectedLabels) {
-		t.Errorf("should invoke ExecuteCallback(Labels=%v) but invoked with %v", expectedLabels, callbackInvoker.Labels)
-	}
-}
```
Comment (bparees) [inline pkg/build/strategies/sti/postexecutorstep_test.go]: is there any testing left for the invokecallback functionality?

Comment (php-coder) [inline pkg/build/strategies/sti/postexecutorstep_test.go]: Yes, and these tests are failing now. I'm going to update them.

**Comment by php-coder** [comment]
@bparees Overall it's OK?

**Comment by bparees** [comment]
> @bparees Overall it's OK?

as long as something is still testing the callback functionality, yes

**Comment by php-coder** [comment]
@bparees PTAL again.

**Comment by bparees** [comment]
[merge]

**Pull request merged**